src/test/java/com/google/javascript/clutz/OptionsTest.java

```diff
@@ -146,4 +146,20 @@ public void testGoogProvides() throws Exception {
             });
     assertThat(opts.knownGoogProvides).containsExactly("foo.bar", "baz.quux");
   }
+
+  @Test
+  public void testPartialInputIgnoresDepgraphRootExternsIfNotPassedToExterns() throws Exception {
+    // Due to "exported" libraries, what the depgraph considers "root" can be incorrect for the
+    // purposes of incremental clutz. Arguments and Externs lists should only be filtered down
+    // with depgraph info and never extended.
+    Options opts =
+        new Options(
+            new String[] {
+              "--partialInput",
+              "javascript/common/dom.js",
```

Comment:
You've removed this from the depgraph - I think you can just replace with foo.js.